+adi/+AD2S1210/Rx.m

```diff
@@ -0,0 +1,123 @@
+        %   integer.
+        SamplesPerFrame = 4096
+    end
+
+    properties
+        % Angle
+        %   Resolver angle in Degrees.
+        Angle = 0
+
+        % AngleScale Angular Scale
+        %   Resolver angle scale.
+        AngleScale = 0
+
+        % Velocity Angular Velocity
+        %   Resolver velocity in revolutions per second.
+        Velocity = 0
+
+        % VelocityScale Velocity Angular Scale
+        %   Resolver velocityscale.
+        VelocityScale = 0
+
+        % ExcitationFrequency Excitation Frequency
+        %   Resolver excitation frequency in Hertz.
+        ExcitationFrequency = 0
+    end
+
+    properties (Hidden)
+        % Number of frames or buffers of data to capture
+        FrameCount = 1
+    end
+
+    % Channel names
+    properties (Nontunable, Hidden, Constant)
+        channel_names = {'angl0', 'anglvel0'}
+    end
+
+    % isOutput
+    properties (Hidden, Nontunable, Access = protected)
+        isOutput = false
+    end
+
+    properties (Nontunable, Hidden)
+        Timeout = Inf
+        kernelBuffersCount = 2
+        dataTypeStr = 'int16'
+        phyDevName = 'ad2s1210'
+        devName = 'ad2s1210'
+    end
+
+    properties (Nontunable, Hidden, Constant)
+        Type = 'Rx'
+    end
+
+    properties (Hidden, Constant)
+        ComplexData = false
+    end
+
+    methods
+        %% Constructor
+        function obj = Rx(varargin)
+            obj = obj@matlabshared.libiio.base(varargin{:});
+            obj.enableExplicitPolling = false;
+            obj.EnabledChannels = [1 2];
+            obj.BufferTypeConversionEnable = true;
+        end
+
+        %% Check Angle
+        function rValue = get.Angle(obj)
+            if obj.ConnectedToDevice
+                rValue = obj.getAttributeRAW('angl0', 'raw', obj.isOutput);
+                obj.Angle = rValue;
+            end
+        end
+
+        %% Check Angular Scale
+        function rValue = get.AngleScale(obj)
+            if obj.ConnectedToDevice
+                rValue = obj.getAttributeDouble('angl0', 'scale', obj.isOutput);
+                obj.AngleScale = rValue;
+            end
+        end
+
+        %% Check Velocity
+        function rValue = get.Velocity(obj)
+            if obj.ConnectedToDevice
+                rValue = obj.getAttributeRAW('anglvel0','raw', obj.isOutput);
+                obj.Velocity = rValue;
+            end
+        end
+
+        %% Check Velocity Scale
+        function rValue = get.VelocityScale(obj)
+            if obj.ConnectedToDevice
+                rValue = obj.getAttributeDouble('anglvel0', 'scale', obj.isOutput);
+                obj.VelocityScale = rValue;
+            end
+        end
+    end
+
+    %% API Functions
+    methods (Hidden, Access = protected)
+
+        function setupInit(obj)
+            % Read excitation frequency attribute from device once connected.
+
+            obj.ExcitationFrequency = obj.getDeviceAttributeLongLong('excitation_frequency');
```

Comment:
This should be a write and not a read. MATLAB translates states at initialization based on MATLAB's pre-connection state

Reply:
Reworded "Read" into "Write" in that comment to better reflect what initialization method is doing here.